src/app/toolbar-panel/toolbar-panel.component.ts

```diff
@@ -80,16 +84,35 @@ export class ToolbarPanelComponent implements OnInit, AfterViewInit {
     //   console.log(response);
     // });
 
+    this.areaService.getAreas().then(areas => {
```

Comment:
onInit should have so much of logic itself, better make a new function to fetch areas